pkg/operator/sync.go

```diff
@@ -0,0 +1,255 @@
+		return err
+	}
+	crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
+	_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
+	if err != nil {
+		return err
+	}
+
+	rbBytes, err := render.RenderAsset(&config, "manifests/clusterapi-apiserver-role-binding.yaml")
+	if err != nil {
+		return err
+	}
+	rb := resourceread.ReadRoleBindingV1OrDie(rbBytes)
+	_, _, err = resourceapply.ApplyRoleBinding(optr.kubeClient.RbacV1(), rb)
+	if err != nil {
+		return err
+	}
+
+	svcBytes, err := render.RenderAsset(&config, "manifests/clusterapi-apiserver-svc.yaml")
+	if err != nil {
+		return err
+	}
+	svc := resourceread.ReadServiceV1OrDie(svcBytes)
+	_, _, err = resourceapply.ApplyService(optr.kubeClient.CoreV1(), svc)
+	if err != nil {
+		return err
+	}
+
+	apiServiceBytes, err := render.RenderAsset(&config, "manifests/clusterapi-apiservice.yaml")
+	if err != nil {
+		return err
+	}
+	apiService := resourceread.ReadAPIServiceDefinitionV1Beta1OrDie(apiServiceBytes)
+	_, _, err = resourceapply.ApplyAPIServiceDefinition(optr.apiregistrationClient, apiService)
+	if err != nil {
+		return err
+	}
+
+	controllerBytes, err := render.RenderAsset(&config, "manifests/clusterapi-apiserver.yaml")
+	if err != nil {
+		return err
+	}
+	controller := resourceread.ReadDeploymentV1OrDie(controllerBytes)
+	_, updated, err := resourceapply.ApplyDeployment(optr.kubeClient.AppsV1(), controller)
+	if err != nil {
+		return err
+	}
+	if updated {
+		return optr.waitForDeploymentRollout(controller)
+	}
+	return nil
+}
+
+func (optr *Operator) syncClusterAPIController(config render.OperatorConfig) error {
+	crBytes, err := render.RenderAsset(&config, "manifests/clusterapi-controller-cluster-role.yaml")
+	if err != nil {
+		return err
+	}
+	cr := resourceread.ReadClusterRoleV1OrDie(crBytes)
+	_, _, err = resourceapply.ApplyClusterRole(optr.kubeClient.RbacV1(), cr)
+	if err != nil {
+		return err
+	}
+
+	crbBytes, err := render.RenderAsset(&config, "manifests/clusterapi-controller-cluster-role-binding.yaml")
+	if err != nil {
+		return err
+	}
+	crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
+	_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
+	if err != nil {
+		return err
+	}
+
+	controllerBytes, err := render.RenderAsset(&config, "manifests/clusterapi-controller.yaml")
+	if err != nil {
+		return err
+	}
+	controller := resourceread.ReadDeploymentV1OrDie(controllerBytes)
+	_, updated, err := resourceapply.ApplyDeployment(optr.kubeClient.AppsV1(), controller)
+	if err != nil {
+		return err
+	}
+	if updated {
+		return optr.waitForDeploymentRollout(controller)
+	}
+	return nil
+}
+
+const (
+	deploymentRolloutPollInterval = time.Second
+	deploymentRolloutTimeout      = 5 * time.Minute
+
+	customResourceReadyInterval = time.Second
+	customResourceReadyTimeout  = 5 * time.Minute
+)
+
+func (optr *Operator) waitForCustomResourceDefinition(resource *apiextv1beta1.CustomResourceDefinition) error {
```

Comment:
Why is this needed to check for all the CRDs? Is it just a placeholder and checks will be added to skip non cluster-api CRDs?